cmd/snap-confine/seccomp-support.c

```diff
@@ -109,13 +109,38 @@ static void validate_bpfpath_is_safe(const char *path)
 	}
 }
 
-bool sc_apply_seccomp_profile_for_security_tag(const char *security_tag)
-{
+bool sc_apply_seccomp_profile_for_security_tag(const char *security_tag) {
 	debug("loading bpf program for security tag %s", security_tag);
 
 	char profile_path[PATH_MAX] = { 0 };
-	sc_must_snprintf(profile_path, sizeof(profile_path), "%s/%s.bin",
+
+        struct sock_fprog prog_allow = { 0 };
+	sc_must_snprintf(profile_path, sizeof(profile_path), "%s/%s.bin.allow",
```

Comment:
would it be better to name these ".allow.bin" and ".deny.bin"? (that way we can still have the .src and .bin suffixes)

Reply:
While working on this I noticed that having two files means we are less atomic than before. So I changed the code to use a directory for the binary profiles (with renameat2(RENAME_EXCHANGE) and to called the filters inside now `filter.{allow,deny}`  (but happy to change things as needed).